src/components/side-panel/coline-side-panel/info-section.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Box } from '@chakra-ui/react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../../redux';
+import RmgFields, { RmgFieldsField } from '../../common/rmg-fields';
+import {
+    getPossibleStnIdsFromMainLine,
+    getPossibleStnIdsFromBranchLine,
+    updateColine,
+} from '../../../redux/param/coline-action';
+import { setGlobalAlert } from '../../../redux/app/action';
+
+export default function InfoSection() {
+    const dispatch = useDispatch();
+
+    // TODO: update coline info section when selectedColine changed, should be, but not working now
+    const selectedColine = useAppSelector(state => state.app.selectedColine);
+    const { stn_list: stnList } = useAppSelector(state => state.param);
+    // FIXME: return emepty component if there is no coline
+    const { from, to } = useAppSelector(state => state.param.coline.at(selectedColine))!;
+    const { branches } = useAppSelector(state => state.helper);
+
+    // https://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript
+    // I have no idea why this will complain and the func should be in (...a: string[][]) => string[][]
+    // @ts-ignore
+    const cartesian = (...a: string[][]): string[][] => a.reduce((a, b) => a.flatMap(d => b.map(e => [d, e].flat())));
```

Comment:
Because of missing default value? Or I'm not sure if `.reduce<string[][]>()` helps.